### ISSUE TITLE
v1.2.0 is incompatible with docker v20.10.17

### ISSUE THREAD

**Issue by Justin-W** [oras-project/oras-go#242]
There seems to be a mismatch between the oras-go v1.2.0 `go.mod` and some of the code.

- https://github.com/oras-project/oras-go/blob/v1.2.0/go.mod#L10
  * This requires `github.com/docker/docker v20.10.17+incompatible`
- https://github.com/oras-project/oras-go/blob/v1.2.0/pkg/auth/docker/login_tls.go#L181
  * this line actually uses the `func PingV2Registry` signature matching the [docker v22 API](https://github.com/moby/moby/blob/v22.06.0-beta.0/registry/auth.go#L173)
  * it seems like it _should_ use the `func PingV2Registry` signature matching the [docker v20 API](https://github.com/moby/moby/blob/v20.10.17/registry/auth.go#L203)

As a result, running `go get -u ./...` on my project (which indirectly depends on oras-go via kops v1.24.0 (both kops v1.24.0 and kops latest still vendor github.com/docker/docker v20) results in a compilation error:
```
../../../../../../pkg/mod/oras.land/oras-go@v1.2.0/pkg/auth/docker/login_tls.go:181:30: assignment mismatch: 3 variables but registry.PingV2Registry returns 2 values
```

It seems that either:
1. oras-go v1.2.0 should still be using the docker v20-compatible API [here](https://github.com/oras-project/oras-go/blob/v1.2.0/pkg/auth/docker/login_tls.go#L181); _OR_
2. oras-go v1.2.0 should be depending on a docker v22 version [here](https://github.com/oras-project/oras-go/blob/v1.2.0/go.mod#L10), _**and**_ the [v1.2.0 release notes](https://github.com/oras-project/oras-go/releases/tag/v1.2.0) should have included a breaking change notice about incompatibility with docker v20.

**Comment by shizhMSFT** [comment]
Nice catch. /cc @souleb.

**Comment by shizhMSFT** [comment]
@jasminetMSFT Could you investigate this issue?

**Comment by jasminetMSFT** [comment]
Sure, please assign this to me.

**Comment by jasminetMSFT** [comment]
@Justin-W In oras-go v1.2.0, https://github.com/oras-project/oras-go/blob/v1.2.0/pkg/auth/docker/login_tls.go#L181 `PingV2Registry` returns three parameters which matches `func PingV2Registry` in [docker v20 API](https://github.com/moby/moby/blob/v20.10.17/registry/auth.go#L203) 
You could run `go get -u ./...` and it should work properly.

**Comment by shizhMSFT** [comment]
Marking this issue as resolved as `oras-go` v1.2.0` is indeed using the docker v20 API as verified by @jasminetMSFT.

**Comment by lahabana** [comment]
Will there be a new patch release anytime soon?
